src/components/pages/search.js

```diff
@@ -1,26 +1,49 @@
-import React, { useMemo } from 'react';
+import React, { useMemo, useState } from 'react';
 import Page from '../page.js';
 import PropTypes from 'prop-types';
 import moment from 'moment';
 import firstBy from 'thenby';
-
-import { Table, Tag } from 'antd';
+import { SearchOutlined } from '@ant-design/icons';
+import { Table, Tag, Select, Form } from 'antd';
 import { useTranslation } from 'react-i18next';
 import { searchResultShape } from '../../ui/default-prop-types.js';
 import { useLanguage } from '../language-context.js';
+import { useRequest } from '../request-context.js';
 import urls from '../../utils/urls.js';
 
 function Search({ initialState }) {
   const { t } = useTranslation('search');
   const { locale } = useLanguage();
   const { docs } = initialState;
+  const { query } = useRequest();
+
   const sortedDocs = useMemo(
-    () => docs.sort(firstBy(doc => doc.contributors)
-      .thenBy(doc => doc.updatedOn, 'desc')),
+    () => docs
+      .map(doc => ({
+        ...doc,
+        tagsSet: new Set(doc.tags)
```

Comment:
Can it be that a document has duplicated tags? 🤔

Reply:
The set is used for a O(1) search during counting. 

Old documents have duplicate tags because of the way that they were generated unfortunately.

Reply:
I saw where it was used. Ok so there can be duplicate tags, that was my question, otherwise the code would have been a bit misleadin